sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/implementation/FileSystemsImpl.java

```diff
@@ -1125,15 +1176,19 @@ public Mono<Response<Void>> getPropertiesNoCustomHeadersWithResponseAsync(String
     public ResponseBase<FileSystemsGetPropertiesHeaders, Void> getPropertiesWithResponse(String requestId,
         Integer timeout, Context context) {
         final String accept = "application/json";
-        return service.getPropertiesSync(this.client.getUrl(), this.client.getFileSystem(), this.client.getResource(),
-            requestId, timeout, this.client.getVersion(), accept, context);
+        try {
+            return service.getPropertiesSync(this.client.getUrl(), this.client.getFileSystem(),
+                this.client.getResource(), requestId, timeout, this.client.getVersion(), accept, context);
+        } catch (DataLakeStorageExceptionInternal internalException) {
+            throw (DataLakeStorageException) ModelHelper.mapToDataLakeStorageException(internalException);
```

Comment:
Why is this casting necessary? Can we update the utility method to just return DataLakeStorageException always?

Reply:
I can make this change if we only ever accept `*Internal` exceptions. Will need to update the asynchronous mapping code.